src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -2,18 +2,90 @@
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final int INITIAL_DEFAULT_CAPACITY = 16;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int MULTIPLICATION_FACTOR = 2;
+    private int size;
+    private Node<K, V>[] table;
+
+    MyHashMap() {
+        table = new Node[INITIAL_DEFAULT_CAPACITY];
+    }
+
     @Override
     public void put(K key, V value) {
+        if (size > table.length * DEFAULT_LOAD_FACTOR) {
+            resize();
+        }
+        int bucket = getHash(key);
+        Node<K, V> currentNode = table[bucket];
+        Node<K, V> newNode = new Node<>(key, value, null);
+
+        while (currentNode != null) {
+            if (currentNode.key == key || key != null && key.equals(currentNode.key)) {
+                currentNode.value = value;
+                return;
+            }
+            if (currentNode.next == null) {
+                currentNode.next = newNode;
+                size++;
+                return;
+            }
+            currentNode = currentNode.next;
+        }
+        table[bucket] = newNode;
+        size++;
 
     }
 
     @Override
     public V getValue(K key) {
+
+        if (size == 0) {
+            return null;
+        }
+        int bucket = getHash(key);
+        Node<K, V> node = table[bucket];
+        while (node != null) {
+            if (key == null && node.key == null || key != null && key.equals(node.key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+
+        return size;
+    }
+
+    private int getHash(K key) {
+        return (key == null ? 0 : Math.abs(key.hashCode() % table.length));
+    }
+
+    private void resize() {
+        Node<K, V>[] oldTable = table;
+        table = new Node[oldTable.length * MULTIPLICATION_FACTOR];
```

Comment:
Here again, avoid using raw types. Use the diamond operator: `table = new Node<>[oldTable.length * MULTIPLICATION_FACTOR];`